src/foam/core/ClassInfoImpl.java

```diff
@@ -72,6 +72,9 @@ public ClassInfo addAxiom(Axiom a) {
     if ( a instanceof PropertyInfo ) {
       ((PropertyInfo)a).setClassInfo(this);
     }
+    if ( a instanceof MethodInfo) {
```

Comment:
In my humble opinion, I am not sure that an interface is necessary! Apparently not all axioms are ClassInfoAware. So maybe it would be added complexity for not much if we added another level. To be discussed?!

Reply:
>Apparently not all axioms are ClassInfoAware.
This is why we need the extra interface. Otherwise, we're going to have to add code here every time we create a new axiom type that needs the class info set. If all axioms were ClassInfoAware then we could blindly set the class info on all axioms that get added here but since that's not the case, we need some other way of identifying that an axiom can have the class info set on it.

Reply:
Ok done